### ISSUE TITLE
Feature: JSON output

### ISSUE THREAD

**Issue by carbontwelve** [tomloprod/time-warden#6]
This looks like an impressive tool. As well as the human readable result produced by the `output` function I would like to also see `toArray` and `toJson` methods for programatic export, e.g to an upstram such as Grafana.

If I were to refactor I would move the majority of `TimeWardenManager::output` into a new `TimeWardenManager::toArray` method that both `TimeWardenManager::output` and a new `TimeWardenManager::toJson` method consume in order to provide a range of export options.

My use case is benchmarking certain code paths within a large Laravel application and needing to collect agrigate data over a period of days in order to identify the cause of a rare but performance impacting issue.

**Comment by tomloprod** [comment]
Hello @carbontwelve I'm glad you find the library useful, it was made with love 🙂

I like what you're proposing! Currently, you can obtain this information through the `getGroups()` and `getTasks()` methods, but it's impractical to repeatedly loop through them.

If you feel like making a PR with your idea, go ahead :-) if not, I might do it myself at some point.

**Comment by tomloprod** [comment]
Good morning! I’ve just released version `1.2.1` of TimeWarden, where you can obtain a summary of all the groups and tasks you’ve created with TimeWarden.

You can do this as follows:

```php
timeWarden()->getSummary();
```
This will return a `TimeWardenSummary` object, which has two methods: `toJson()` and `toArray()`.

Here is an example of the data returned in array format:

```php
    $summaryArray = [
        [
            'name' => 'default',
            'duration' => 42.0,
            'tasks' => [
                [
                    'name' => 'TaskName1',
                    'duration' => 19.0,
                    'friendly_duration' => '19ms',
                    'start_timestamp' => 1496664000.0,
                    'end_timestamp' => 1496664000.019,
                    'start_datetime' => '2017-06-05T12:00:00+00:00',
                    'end_datetime' => '2017-06-05T12:00:00+00:00',
                ],
                [
                    'name' => 'TaskName2',
                    'duration' => 23.0,
                    'friendly_duration' => '23ms',
                    'start_timestamp' => 1496664000.0,
                    'end_timestamp' => 1496664000.023,
                    'start_datetime' => '2017-06-05T12:00:00+00:00',
                    'end_datetime' => '2017-06-05T12:00:00+00:00',
                ],
            ],
        ],
        [
            // Others groups...
        ]
    ];
];
```

I hope you find it useful. Have a great Sunday!
